### PR TITLE
DAOS-623 build: Use leap-dnf Docker image (#13271) (#13311)

### PR DESCRIPTION
**Pull request by brianjmurrell**
Rather than using zypper to bootstrap dnf, use an image that has dnf
built in.

Remove version from go packages on Leap 15 to ensure we are alwys using
the latest version availalble for SDL requirements.

Add targets to our EL8 and Leap 15 Dockerfiles to allow building more
basic containers that are just configured to use our Artifactory repo
server.  This can be utilized by adding "--target basic" to your docker/
podman build command.

Sync EL9 Dockerfile to EL8 and Leap15 format.

Fix for OneAPI compilter packaging changes

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: site_scons/site_tools/doneapi.py [hunk 1]
```diff
@@ -18,22 +18,19 @@ class DetectCompiler():
...
     def __init__(self):
         root = '/opt/intel/oneapi/compiler/latest'
-        binp = os.path.join(root, 'linux', 'bin')
-        libp = os.path.join(root, 'linux', 'lib')
-        include = os.path.join(root, 'linux', 'include')
-        binarch = os.path.join(binp, 'intel64')
-        libarch = os.path.join(root, 'linux', 'compiler', 'lib', 'intel64_lin')
+        binp = os.path.join(root, 'bin')
+        libp = os.path.join(root, 'lib')
+        include = os.path.join(root, 'include')
```
Comment (daltonbohning) [inline site_scons/site_tools/doneapi.py:23]: I just want to point out that this is not backwards compatible with older versions, right?

**Review by daltonbohning — DISMISSED** [review]

File: site_scons/site_tools/doneapi.py (hunk 1 above)
Comment (brianjmurrell) [reply site_scons/site_tools/doneapi.py:23]: That's correct.  If we have a use-case for using an older version, I suppose some backward compatibility could be added.

Comment (ashleypittman) [reply site_scons/site_tools/doneapi.py:23]: I think the question to ask here is why do we have icc support in scons and the answer is not so we can build with icc in CI, the answer is that we want users to be able to use icc with daos and therefore it should be obvious that backwards compatibility is desirable here.  Like Brian says something to address in master however.

Comment (brianjmurrell) [reply site_scons/site_tools/doneapi.py:23]: But surely everyone would want to be using the most recent icc wouldn't they?

Comment (ashleypittman) [reply site_scons/site_tools/doneapi.py:23]: want to certainly, but as we've demonstrated they might not be and may want to be updating daos without forcing a system software change.

**Pull request merged**